src/Compilers/CSharp/Portable/Symbols/Source/CustomModifierUtils.cs

```diff
@@ -74,18 +74,17 @@ internal static TypeSymbol CopyTypeCustomModifiers(TypeSymbol sourceType, TypeSy
             // code gen uses and then passing them to the dynamic type decoder that metadata reading uses.
             // NOTE: ref is irrelevant here since we are just encoding/decoding the type out of the signature context
             ImmutableArray<bool> flags = CSharpCompilation.DynamicTransformsEncoder.EncodeWithoutCustomModifierFlags(destinationType, refKind);
-            TypeSymbol typeWithDynamic = DynamicTypeDecoder.TransformTypeWithoutCustomModifierFlags(sourceType, containingAssembly, refKind, flags);
+            TypeSymbol resultType = DynamicTypeDecoder.TransformTypeWithoutCustomModifierFlags(sourceType, containingAssembly, refKind, flags);
+
+            var builder = ArrayBuilder<bool>.GetInstance();
```

Comment:
nit: consider adding an overload of `Encode` that returns flags as an immutable array (similarly to dynamic and tuple decoders)

Reply:
This would be the only use of that overload. Let's add later if there are other uses.

---
In reply to: [485231913](https://github.com/dotnet/roslyn/pull/47536#discussion_r485231913) [](ancestors = 485231913)